File: src/main/scala/im/tox/toktok/app/MainChatsFragment/ChatsFragment.scala
package im.tox.toktok.app.MainChatsFragment

import android.graphics.Color
import android.os.Bundle
import android.support.v4.app.Fragment
import android.support.v7.app.AppCompatActivity
import android.support.v7.widget.{LinearLayoutManager, RecyclerView}
import android.view.{LayoutInflater, View, ViewGroup}
import im.tox.toktok.R
import im.tox.toktok.app.MainFriendsFragment.Friend

import scala.collection.mutable.ListBuffer

class ChatsFragment extends Fragment {

  var mChats_Recycler: RecyclerView = null

  override def onCreateView(inflater: LayoutInflater, container: ViewGroup, savedState: Bundle): View = {

    val view: View = inflater.inflate(R.layout.home_chats, container, false)
    val activity: AppCompatActivity = getActivity.asInstanceOf[AppCompatActivity]

    // Recycler View

    mChats_Recycler = view.findViewById(R.id.home_chats_recycler).asInstanceOf[RecyclerView]

    val mLayoutManager: LinearLayoutManager = new LinearLayoutManager(activity.getBaseContext)
    mChats_Recycler.setLayoutManager(mLayoutManager)
    var a = ListBuffer[ChatsMessageObject]()

    a += new ChatsMessageObject(new Friend("Lorem Ipsum", "Trying to TokTok", 0, Color.parseColor("#DC4254"), Color.parseColor("#7D2530"), R.drawable.lorem), "Hello, how are you?")
    a += new ChatsMessageObject(new Friend("John Doe", "Up!", 0, Color.parseColor("#009688"), Color.parseColor("#00796B"), R.drawable.john), "Hello, how are you?")

    var friends = ListBuffer[Friend]()

    friends += new Friend("Lorem Ipsum", "Trying to TokTok", 0, Color.parseColor("#DC4254"), Color.parseColor("#7D2530"), R.drawable.lorem)
    friends += new Friend("John Doe", "Up!", 0, Color.parseColor("#009688"), Color.parseColor("#00796B"), R.drawable.john)

    a += new ChatsMessageObject(new Group("The Amazing Group", friends, Color.parseColor("#9B9B9B"), Color.parseColor("#5A5A5A")), "Let's Go!")


    val mChats_Recycler_Adapter: ChatsRecyclerAdapter = new ChatsRecyclerAdapter(a)
    mChats_Recycler.setAdapter(mChats_Recycler_Adapter)

    return view

  }

}

File: sources/app/src/main/java/im/tox/toktok/app/main/chats/ChatsFragment.java
package im.tox.toktok.app.main.chats;

import android.content.Intent;
import android.content.res.ColorStateList;
import android.os.Bundle;
import android.os.Handler;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import com.google.android.material.appbar.AppBarLayout;
import com.google.android.material.floatingactionbutton.FloatingActionButton;
import androidx.fragment.app.Fragment;
import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.view.ActionMode;
import androidx.recyclerview.widget.DefaultItemAnimator;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.Menu;
import android.view.MenuItem;
import android.view.View;
import android.view.ViewGroup;
import android.view.animation.AccelerateInterpolator;
import android.view.animation.DecelerateInterpolator;
import android.widget.FrameLayout;

import java.util.Arrays;
import java.util.List;

import im.tox.toktok.R;
import im.tox.toktok.app.CompatUtil;
import im.tox.toktok.app.CustomViewPager;
import im.tox.toktok.app.MyRecyclerScroll;
import im.tox.toktok.app.domain.ChatMessage;
import im.tox.toktok.app.new_message.NewMessageActivity;

public final class ChatsFragment extends Fragment implements ChatItemClick {

    @Nullable
    private ChatsRecyclerAdapter mChatsRecyclerAdapter = null;
    @Nullable
    private ActionMode mActionMode = null;

    @Override
    public FrameLayout onCreateView(@NonNull LayoutInflater inflater, ViewGroup container, Bundle savedState) {
        FrameLayout view = (FrameLayout) inflater.inflate(R.layout.fragment_home_chats, container, false);
        final FloatingActionButton fab = getActivity().findViewById(R.id.home_fab);

        // Recycler View

        RecyclerView mChatsRecycler = view.findViewById(R.id.home_chats_recycler);

        LinearLayoutManager mLayoutManager = new LinearLayoutManager(getActivity().getBaseContext());
        mChatsRecycler.setLayoutManager(mLayoutManager);

        List<ChatMessage> chatMessages = Arrays.asList(
                ChatMessage.loremMessage,
                ChatMessage.johnMessage,
                ChatMessage.groupMessage
        );

        mChatsRecyclerAdapter = new ChatsRecyclerAdapter(chatMessages, this);

        mChatsRecycler.setAdapter(mChatsRecyclerAdapter);
        mChatsRecycler.setItemAnimator(new DefaultItemAnimator());
        mChatsRecycler.addOnScrollListener(new MyRecyclerScroll() {
            @Override
            public void hide() {
                fab.animate().translationY(fab.getHeight() + fab.getBottom()).setInterpolator(new AccelerateInterpolator(2)).start();
            }

            @Override
            public void show() {
                fab.animate().translationY(0).setInterpolator(new DecelerateInterpolator(2)).start();
            }
        });

        return view;
    }

    private void toggleSelection(int i) {
        mChatsRecyclerAdapter.toggleSelection(i);

        int count = mChatsRecyclerAdapter.getSelectedItemCount();
        switch (count) {
            case 0:
                mActionMode.finish();
                break;
            case 1:
                mActionMode.setTitle(getResources().getString(R.string.action_mode_selected_single));
                break;
            default:
                mActionMode.setTitle(count + " " + getResources().getString(R.string.action_mode_selected_multi));
        }
    }

    @Override
    public boolean onClick(int i) {
        if (mActionMode == null) {
            return false;
        }
        toggleSelection(i);
        return true;
    }

    @Override
    public boolean onLongClick(int i) {
        if (mActionMode == null) {
            mActionMode = ((AppCompatActivity) getActivity()).startSupportActionMode(new ChatsActionModeCallback());
        }

        toggleSelection(i);
        return true;
    }

    final class ChatsActionModeCallback implements ActionMode.Callback {
        @Nullable
        private AppBarLayout mAppLayout = null;
        @Nullable
        private FloatingActionButton mFab = null;
        @Nullable
        private CustomViewPager mCustomViewPager = null;

        @Override
        public boolean onCreateActionMode(@NonNull final ActionMode mode, Menu menu) {
            final AppCompatActivity activity = (AppCompatActivity) getActivity();
            mAppLayout = activity.findViewById(R.id.appBarLayout);
            mAppLayout.setBackgroundColor(CompatUtil.getColor(getResources(), R.color.backgroundColor));

            mFab = activity.findViewById(R.id.home_fab);
            mFab.setBackgroundTintList(ColorStateList.valueOf(CompatUtil.getColor(getResources(), R.color.textDarkColor)));
            mFab.setImageResource(R.drawable.ic_action_delete);
            mFab.setImageTintList(ColorStateList.valueOf(CompatUtil.getColor(getResources(), R.color.textWhiteColor)));

            mCustomViewPager = activity.findViewById(R.id.home_tab_holder);
            mCustomViewPager.setSwipingEnabled(false);

            mFab.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    mChatsRecyclerAdapter.deleteSelected();

                    new Handler().postDelayed(new Runnable() {
                        @Override
                        public void run() {
                            mode.finish();
                        }
                    }, 500);
                }
            });

            return true;
        }

        @Override
        public void onDestroyActionMode(ActionMode mode) {
            mAppLayout.setBackgroundColor(CompatUtil.getColor(getResources(), R.color.homeColorToolbar));

            mFab.setImageResource(R.drawable.ic_content_add_home);
            mFab.setBackgroundTintList(ColorStateList.valueOf(CompatUtil.getColor(getResources(), R.color.basicFABColor)));
            mFab.setImageTintList(ColorStateList.valueOf(CompatUtil.getColor(getResources(), R.color.basicFABTint)));

            mFab.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    startActivity(new Intent(getActivity(), NewMessageActivity.class));
                }
            });

            mCustomViewPager.setSwipingEnabled(true);

            mChatsRecyclerAdapter.clearSelections();
            mActionMode = null;
            mAppLayout = null;
            mFab = null;
            mCustomViewPager = null;
        }

        @Override
        public boolean onActionItemClicked(ActionMode mode, MenuItem item) {
            return true;
        }

        @Override
        public boolean onPrepareActionMode(ActionMode mode, Menu menu) {
            return false;
        }

    }

}
